hasher.go

```diff
@@ -316,3 +302,24 @@ func min(ns []byte, ns2 []byte) []byte {
 	}
 	return ns2
 }
+
+func computeNsRange(leftMinNs, leftMaxNs, rightMinNs, rightMaxNs []byte, ignoreMaxNs bool, precomputedMaxNs namespace.ID) (minNs []byte, maxNs []byte) {
```

Comment:
This function contains the new version of ns range calculation for a parent node.